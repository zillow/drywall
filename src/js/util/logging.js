export const log = (message, level = 'warn') => {
    if (typeof process !== 'undefined' && process.env.NODE_ENV !== 'production') {
        // eslint-disable-next-line no-console
        console[level](message);
    }
};

export const logDeprecation = message => {
    log(`[Deprecation] ${message}`);
};
