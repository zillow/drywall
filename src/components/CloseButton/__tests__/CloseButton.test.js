import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider, css, isStyledComponent } from 'styled-components';
import { CloseButton } from '../../../index';
import { createTestTheme } from '../../../test/util';

describe('<CloseButton>', () => {
    it('can render without theme', () => {
        const testRenderer = TestRenderer.create(<CloseButton />);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });
    it('can render with theme', () => {
        const theme = createTestTheme({
            CloseButton: css`
                color: red;
            `,
        });
        const testRenderer = TestRenderer.create(
            <ThemeProvider theme={theme}>
                <CloseButton />
            </ThemeProvider>
        );
        const tree = testRenderer.toJSON();
        expect(tree).toHaveStyleRule('color', 'red');
        expect(tree).toMatchSnapshot();
    });
    it('is a styled-component', () => {
        expect(isStyledComponent(CloseButton)).toBe(true);
    });
    it('supports the as prop', () => {
        const testRenderer = TestRenderer.create(<CloseButton as="a" />);
        const tree = testRenderer.toJSON();
        expect(tree.type).toBe('a');
        expect(tree).toMatchSnapshot();
    });
});
