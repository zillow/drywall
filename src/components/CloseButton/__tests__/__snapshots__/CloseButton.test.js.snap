// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton> can render with theme 1`] = `
.c0 {
  color: red;
}

<button
  className="c0"
/>
`;

exports[`<CloseButton> can render without theme 1`] = `
<button
  className="CloseButton-sqbalp-0 "
/>
`;

exports[`<CloseButton> supports the as prop 1`] = `
<a
  className=""
/>
`;
